packages/typechain/src/parser/normalizeName.ts

```diff
@@ -1,16 +1,40 @@
-import { upperFirst, camelCase } from 'lodash'
+import { upperFirst } from 'lodash'
 
 /**
  * Converts valid file names to valid javascript symbols and does best effort to make them readable. Example: ds-token.test becomes DsTokenTest
  */
 export function normalizeName(rawName: string): string {
-  const t1 = rawName.split(' ').join('-') // spaces to - so later we can automatically convert them
-  const t2 = t1.replace(/^\d+/, '') // removes leading digits
-  const result = upperFirst(camelCase(t2))
+  const transformations: ((s: string) => string)[] = [
+    (s) => s.split(' ').join('-'), // spaces to - so later we can automatically convert them
+    (s) => s.replace(/^\d+/, ''), // removes leading digits
+    (s) => deleteCharacterAndCaptializeNextCharacter(s, '-'),
+    (s) => deleteCharacterAndCaptializeNextCharacter(s, '_'),
+    (s) => deleteCharacterAndCaptializeNextCharacter(s, '.'),
```

Comment:
another idea: replace all non-word char sequences with a single '-' in one go, so you don't have to repeat the bottom lines:
```suggestion
    (s) => s.replace(/\W+/g, '-'), // replace non-word character sequences with a single '-'
    (s) => s.replace(/^\d+/, ''), // removes leading digits
    (s) => deleteCharacterAndCaptializeNextCharacter(s, '-'),
```

Reply:
... and yet another option without the need for a helper function :D 
```suggestion
    (s) => s.replace(/^\d+/, ''), // remove leading digits
    (s) => s.replace(/\W+/g, '-'), // replace non-word character sequences with a single '-'
    (s) => s.replace(/-[a-z]/g, match => match.substr(-1).toUpperCase()), // delete '-' and capitalize the letter after them
    (s) => s.replace(/-/g, ''), // remove remaining minuses
```